### PR TITLE
issue #29137 Allow saving reports to different packages

### PR DESCRIPTION
**Pull request by ChristopherCotnoir**

### CODE REVIEW

File: OpenRPT/wrtembed/dbfiledialog.cpp [hunk 1]
```diff
@@ -39,20 +39,30 @@ DBFileDialog::DBFileDialog(QWidget* parent, Qt::WindowFlags fl)
...
 
     XSqlQuery qry(getSqlFromTag("fmt12", QSqlDatabase::database().driverName()));
     while(qry.next())
     {
-      QTreeWidgetItem * item = new QTreeWidgetItem(_list, QStringList() << qry.value("report_name").toString() << qry.value("report_grade").toString());
+      QTreeWidgetItem * item = new QTreeWidgetItem(_list, QStringList() << qry.value("report_name").toString() << qry.value("report_grade").toString() << qry.value("package").toString());
       item->setData(0, Qt::UserRole, qry.value("report_id"));
     }
+
+    _package->addItem("");
```
Comment (gilmoskowitz) [inline OpenRPT/wrtembed/dbfiledialog.cpp]: This extra line is unnecessary if the `fmt21` query gets simplified and the `_packageLit` label text changed to "Schema".

File: OpenRPT/wrtembed/dbfiledialog.cpp [hunk 2]
```diff
@@ -128,32 +143,54 @@ int DBFileDialog::getGradeById()
   return -1;
 }
 
+QString DBFileDialog::getPackageById()
+{
+  int rid = getId();
+  if(rid != -1)
+  {
+          XSqlQuery src_qry;
+    src_qry.prepare(getSqlFromTag("fmt20", QSqlDatabase::database().driverName())); // MANU
```
Comment (gilmoskowitz) [inline OpenRPT/wrtembed/dbfiledialog.cpp]: No need for the `MANU` comment. I think he was a contributor a few years ago.

File: OpenRPT/wrtembed/dbfiledialog.cpp [hunk 3]
```diff
@@ -128,32 +143,54 @@ int DBFileDialog::getGradeById()
...
+    src_qry.bindValue(":report_id", rid);
+    src_qry.exec();
+          if(src_qry.first())
+            return src_qry.value("package").toString();
+  }
+  return QString::null;
+}
```
Comment (gilmoskowitz) [inline OpenRPT/wrtembed/dbfiledialog.cpp]: Watch indentation

File: OpenRPT/wrtembed/dbfiledialog.cpp [hunk 4]
```diff
@@ -128,32 +143,54 @@ int DBFileDialog::getGradeById()
...
+void DBFileDialog::sPackageChanged( const QString & )
+{
+  sNameGradePackageChanged();
 }
 
-void DBFileDialog::sNameGradeChanged()
+void DBFileDialog::sNameGradePackageChanged()
 {
   QString name  = getName();
   QString grade = QString("%1").arg(getGrade());
```
Comment (gilmoskowitz) [inline OpenRPT/wrtembed/dbfiledialog.cpp]: Simplify with [QString::number()](http://doc.qt.io/qt-5/qstring.html#number-2)?

**Comment by ChristopherCotnoir** [comment]
Gil and I have determined that the handling for database servers is already broken significantly enough that adding support here wouldn't do any good.

**Comment by gilmoskowitz** [comment]
The CI failure is the result of a difference between Qt4 and Qt5. We've dealt with it [elsewhere with `#if`](https://github.com/xtuple/qt-client/blob/4_11_x/guiclient/characteristic.cpp#L383-L387).

**Pull request merged**